DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/ViewDelegateController.cs

```diff
@@ -48,5 +53,32 @@ public IActionResult Index(int delegateId)
             var model = new ViewDelegateViewModel(delegateInfoViewModel, courseInfoViewModels);
             return View(model);
         }
+
+        [Route("SendWelcomeEmail")]
+        public IActionResult SendWelcomeEmail(int delegateId)
+        {
+            var centreId = User.GetCentreId();
+
+            var delegateUser = userDataService.GetDelegateUserCardById(delegateId);
+            if (delegateUser == null || delegateUser.CentreId != centreId)
+            {
+                return new NotFoundResult();
+            }
+
+            registrationService.GenerateAndSendDelegateWelcomeEmail(delegateUser!);
+
+            TempData.Set(new WelcomeEmailSentViewModel(delegateUser));
+
+            return RedirectToAction("WelcomeEmailSent", new { delegateId = delegateUser.Id });

```

Comment:
Could we just show the successful View here instead of dealing with TempData & redirects?